sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -59,13 +59,12 @@
 class RunnerIOOperation(operations.Operation):
   """Common baseclass for runner harness IO operations."""
 
-  def __init__(self, operation_name, step_name, consumers, counter_factory,
+  def __init__(self, name_context, step_name, consumers, counter_factory,
                state_sampler, windowed_coder, target, data_channel):
     super(RunnerIOOperation, self).__init__(
-        operation_name, None, counter_factory, state_sampler)
+        name_context, None, counter_factory, state_sampler)
     self.windowed_coder = windowed_coder
     self.windowed_coder_impl = windowed_coder.get_impl()
-    self.step_name = step_name
```

Comment:
Is this deletion intentional?  If so, can you add a comment / JIRA reference to clean up step_name in the arguments?

Reply:
This is part of my goal with BEAM-4028. The step name is meant to only be retrievable through the name context.